pkg/antctl/raw/check/cluster/command.go

```diff
@@ -0,0 +1,230 @@
+// Copyright 2024 Antrea Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cluster
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/fatih/color"
+	"github.com/spf13/cobra"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/utils/ptr"
+
+	"antrea.io/antrea/pkg/antctl/raw/check"
+	"antrea.io/antrea/pkg/version"
+)
+
+func Command() *cobra.Command {
+	command := &cobra.Command{
+		Use:   "cluster",
+		Short: "Runs pre installation checks",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return Run()
+		},
+	}
+	return command
+}
+
+const (
+	testNamespacePrefix = "antrea-test"
+	deploymentName      = "cluster-checker"
+	podReadyTimeout     = 1 * time.Minute
+)
+
+type uncertainError struct {
+	reason string
+}
+
+func (e uncertainError) Error() string {
+	return e.reason
+}
+
+func newUncertainError(reason string, a ...interface{}) uncertainError {
+	return uncertainError{reason: fmt.Sprintf(reason, a...)}
+}
+
+type Test interface {
+	Run(ctx context.Context, testContext *testContext) error
+}
+
+var testsRegistry = make(map[string]Test)
+
+func RegisterTest(name string, test Test) {
+	testsRegistry[name] = test
+}
+
+type testContext struct {
+	client      kubernetes.Interface
+	config      *rest.Config
+	clusterName string
+	namespace   string
+	testPod     *corev1.Pod
+}
+
+func Run() error {
+	client, config, clusterName, err := check.NewClient()
+	if err != nil {
+		return fmt.Errorf("unable to create Kubernetes client: %s", err)
+	}
+	ctx := context.Background()
+	testContext := NewTestContext(client, config, clusterName)
+	if err := testContext.setup(ctx); err != nil {
+		return err
+	}
+	var numSuccess, numFailure, numUncertain int
+	for name, test := range testsRegistry {
+		testContext.Header("Running test: %s", name)
+		if err := test.Run(ctx, testContext); err != nil {
+			if errors.As(err, new(uncertainError)) {
+				testContext.Warning("Test %s was uncertain: %v", name, err)
+				numUncertain++
+			}
+			testContext.Fail("Test %s failed: %v", name, err)
+			numFailure++
```

Comment:
This would result in duplicate result (one warning and one error) when there is an uncertain error.

Reply:
Thanks. changes are pushed